core/amber/src/main/python/core/python_worker.py

```diff
@@ -30,13 +52,53 @@ def run(self) -> None:
             target=self._network_sender.run, name="network_sender"
         )
         main_loop_thread = Thread(target=self._main_loop.run, name="main_loop_thread")
+        self.original_parent_pid = os.getppid()
 
+        loop = asyncio.new_event_loop()
+        heartbeat_thread = Thread(
+            target=self.start_asyncio_loop, name="heartbeat_thread", args=(loop, 5.0)
+        )
```

Comment:
I suggest we let this 5.0 to be configurable. pass it in `python_udf.conf` and let JVM to pass the value here.